palantirnet/st-bernards
# MSSB-35: Update Drupal core to 8.8

## JIRA Ticket(s)

- See ticket [MSSB-35: Update Drupal core to 8.8](https://palantir.atlassian.net/browse/MSSB-35).

## Description

- Updates Drupal core to 8.8.1
- Updates The Build to 2.2.1
- Changes the site profile from `standard` to `config_installer` to allow both `drush cim` and `phing install` to work.

*Note: `phing load` will fail during testing (due to the mismatched profile in the database), but once Prod is updated using the deployment steps, database backups will work correctly.*

## Testing instructions

- `composer install`
- Place a prod database backup in `artifacts/backups`
- `drush sql-drop`
- `phing load` (This will import the database, run `drush updb`, and import config.) You will see it fail with the following error:
```
 [error]  Drupal\Core\Config\ConfigImporterException: There were errors validating the config synchronization.
Cannot change the install profile from <em class="placeholder">standard</em> to <em class="placeholder">config_installer</em> once Drupal is installed.
Unable to install the <em class="placeholder">config_installer</em> module since it does not exist.
Unable to uninstall the <em class="placeholder">Standard</em> profile since it is the install profile. in Drupal\Core\Config\ConfigImporter->validate() (line 755 of /var/www/st-bernards.local/web/core/lib/Drupal/Core/Config/ConfigImporter.php). 
```
- `drush cset core.extension profile config_installer`
- `echo "delete from key_value where collection = 'state' and name = 'system.profile.files';" | drush sqlc`
- `drush cr`
- `drush en config_installer`
- `drush cim -y`
- `drush cr`
- `drush cim -y` to enable the development modules/config
- `drush uublk developer`
- `drush uli`
- Verify the site works as expected.
- Go to https://st-bernards.local/admin/reports/updates and verify the core version is 8.8.1.
- `phing install`
- Verify the site installs cleanly and works properly.

## Deployment instructions

- `drush cr`
- `drush updb`
- `drush cset core.extension profile config_installer`
- `echo "delete from key_value where collection = 'state' and name = 'system.profile.files';" | drush sqlc`
- `drush cr`
- `drush en config_installer`
- `drush cim -y`
